samples/gitchurn/src/gitChurnMain/kotlin/GitChurn.kt

```diff
@@ -57,11 +60,54 @@ private fun calculateChurn(workDir: String, limit: Int) {
         commit.close()
         count++
     }
-    println("Report:")
+    count = 0
+    */
+    println("Calculating by my own way…")
+    myLimited.forEach { commit ->
+        if (count % 100 == 0)
+            println("Commit #$count [${commit.time.format()}] by ${commit.author.name!!.toKString()}: ${commit.summary}")
+
+        commit.parents.forEach { parent ->
+            val diff = commit.tree.diff(parent.tree)
+            diff.deltas().forEach { delta ->
+                //val path = delta.newPath.dropLastWhile { it!='/' }
+                val path = if (delta.newPath == delta.newPath.substringBefore('/'))
+                    delta.newPath
+                else
+                    delta.newPath.substringBefore('/') + "/"
+                val n = namedMap[Pair(parent.author.email!!.toKString().toLowerCase(), path)] ?: 0
+                //map[path] = n + 1
+                namedMap[Pair(parent.author.email!!.toKString().toLowerCase(), path)] = n + 1
+            }
+            diff.close()
+            parent.close()
+        }
+        commit.close()
+        count++
+    }
+
+    /*println("Report:")
     map.toList().sortedByDescending { it.second }.take(10).forEach {
         println("File: ${it.first}")
         println("      ${it.second}")
         println()
+    }*/
+    println("Named Report:")
+
+    namedMap.toList().filter { it.second > 20 }.groupBy { it.first.first }.values.forEach {
```

Comment:
Another reason for using `data class` becomes obvious here:  `it.first.author` is more understandable than `it.first.first`.